python/pypto/language/parser/expr_evaluator.py

```
@@ -138,10 +138,22 @@ def python_value_to_ir(self, value: Any, span: ir.Span) -> ir.Expr:
-        if isinstance(value, Scalar) and not value._annotation_only and value.expr is not None:
-            # Composite over DynVars (e.g. `m + 0`) built via DSL operator
-            # overloading — keep the IR tree as-is (no constant folding).
-            return value.expr
+        if isinstance(value, Scalar) and not value._annotation_only:
+            # Composite over DynVars (e.g. ``m + 0``, ``NR * 64``) built via
+            # DSL operator overloading.  Wrap in DimExpr so the SSA verifier
+            # treats it as a type-level annotation.
+            expr = value.unwrap()
+            if expr is None:
+                raise ParserTypeError(
+                    f"Scalar has no underlying expression",
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)